src/main/java/ui/MenuControl.java

```diff
@@ -161,9 +140,41 @@ private void onBoardSaveAs() {
             String boardName = response.get().trim();
             prefs.addBoard(boardName, panelList);
             prefs.setLastOpenBoard(boardName);
+            prefs.setLastOpenBoardPanelInfo(panelList);
             ui.triggerEvent(new BoardSavedEvent());
             logger.info("New board " + boardName + " saved");
             ui.updateTitle();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Prompt use for whether to save the current board panel info or not
+     * when user is closing current board (i.e. switching to another board)
+     *
+     * @return true if user choose to save, false if user choose to discard
+     */
+    private boolean promptToSaveCurrentBoard() {
+        List<PanelInfo> currentPanelInfo = panels.getCurrentPanelInfos();
+        List<PanelInfo> lastOpenBoardPanelInfo = prefs.getLastOpenBoardPanelInfo();
+
+        if (!currentPanelInfo.equals(lastOpenBoardPanelInfo)) {
+            ConfirmChangesDialog dlg = new ConfirmChangesDialog(mainStage);
+            Optional<ButtonType> response = dlg.showAndWait();
+            prefs.setLastOpenBoardPanelInfo(currentPanelInfo);
+            if (response.get() == ButtonType.OK) {
+                onBoardSave();
+                logger.info("Changes to the current board saved.");
+                return true;
+            } else {
+                logger.info("User abandoned unsaved changes.");
+                return false;
+            }
+        } else {
+            // No changes to current board, no need to save.
+            return false;
```

Comment:
Same here, can put guard clauses first to reduce nesting. Also if the `if` branch ends with a `return` then we don't need an `else` clause.